src/nspawn/nspawn-mount.c

```diff
@@ -250,8 +253,12 @@ int mount_sysfs(const char *dest) {
 
         (void) mkdir(full, 0755);
 
-        r = mount_verbose(LOG_ERR, "sysfs", full, "sysfs",
-                          MS_RDONLY|MS_NOSUID|MS_NOEXEC|MS_NODEV, NULL);
+        if ((mount_settings & MOUNT_RO) > 0) {
+                sysfs_flags |= MS_RDONLY;
+                remount_flags |= MS_RDONLY;
+        }
+
+        r = mount_verbose(LOG_ERR, "sysfs", full, "sysfs", sysfs_flags, NULL);
```

Comment:
I'd really do this the other way round: leave the literal flags parameter being passed to mount_verbose() in place, but add an `|extra` to the ORed flags as last step. Then, simply define `extra` as long, and initialize it based on the `MOUNT_RO` flag, and reuse it for both cases. Such a change would be much shorter, and only needs one extra var, instead of two...